extensions/VisualEditor/wikia/ApiAddMediaPermanent.php

```diff
@@ -41,7 +43,9 @@ private function executeImage() {
 			$file = new LocalFile( $title, RepoGroup::singleton()->getLocalRepo() );
 			$file->upload( $tempFile->getPath(), '', $pageText ? $pageText : '' );
 			return array(
-				'title' => $file->getTitle()->getText()
+				'title' => $file->getTitle()->getText(),
+				'url' => $file->getUrl(),
+				'image_id' => $file->getTitle()->getArticleID()
```

Comment:
Well in MW images do not really have ids - article have ids and there are images associated with those articles based on titles. I'm encouraging you to not create new term - image_id - unless there is a really good reason for it. Other than that this change looks good to me.